### PR TITLE
fix(test): wait for xds_delivery_count to stabilize after each test

### PR DESCRIPTION
**Pull request by lobkovilya**
Instead of just sleeping between tests, we'll sleep starting the moment `xds_delivery_count` is not changing. Each change of the `xds_delivery_count` resets the sleep. This is important for `should deploy mesh wide policy` to not be flaky.

### CODE REVIEW

**Review by jakubdyszkiewicz — APPROVED** [review]

File: test/k8s/simple_test.go [hunk 1]
```diff
@@ -70,7 +70,28 @@ func Simple() {
...
+		Expect(err).ToNot(HaveOccurred())
+
+		stopCh := make(chan struct{})
+		metricCh := make(chan int)
+		errCh := make(chan error)
+
+		go WatchXdsDeliveryCount(promClient, stopCh, metricCh, errCh)
+		defer close(stopCh)
+
+	Loop:
```
Comment (michaelbeaumont) [inline test/k8s/simple_test.go:84]: I don't think I've ever seen a capitalized label before :D

File: test/k8s/simple_test.go [hunk 2]
```diff
@@ -70,7 +70,28 @@ func Simple() {
...
+		for {
+			select {
+			case <-metricCh:
```
Comment (michaelbeaumont) [inline test/k8s/simple_test.go:87]: should this value be used? otherwise might as well be `struct{}`? but doesn't matter

**Review by michaelbeaumont — APPROVED** [review]

File: test/k8s/simple_test.go (hunk 1 above)
Comment (lobkovilya) [reply test/k8s/simple_test.go:84]: It seems like there is no strict guideline on this, internally golang is using both capitalized labels and not. See [examples](https://stackoverflow.com/questions/64039172/whats-the-standard-practice-for-naming-labels-in-go) in the first reply.

Comment (michaelbeaumont) [reply test/k8s/simple_test.go:84]: yeah doesn't matter either way

**Pull request merged**